agent_guard_core/credentials/conjur_secrets_provider.py

```diff
@@ -127,12 +266,48 @@ def _update_token_expiration(self):
     def connect(self) -> bool:
         if not self._access_token or datetime.now(
         ) > self._access_token_expiration:
-            if not self._authenticator_id:
+            if self._authenticator_id.startswith("authn-jwt"):
+                return self._authenticate_jwt()
+            if self._authenticator_id.startswith("authn-iam"):
```

Comment:
Since both conditions can't co exist
```suggestion
            elif self._authenticator_id.startswith("authn-iam"):
```